application/client/src/app/ui/views/workspace/headers/component.ts

```diff
@@ -63,13 +69,44 @@ export class ColumnsHeaders extends ChangesDetector implements AfterContentInit
             this.session.stream.subjects.get().rank.subscribe(() => {
                 this.markChangesForCheck();
             }),
+            this.controller.subjects.get().visibility.subscribe(() => {
+                this.headers = this.controller.headers
+                .filter((h) => h.visible)
+                .map((h) => new RenderedHeader(h));
+                this.markChangesForCheck();
+            })
         );
         this.headers = this.controller.headers
             .filter((h) => h.visible)
             .map((h) => new RenderedHeader(h));
         this.markChangesForCheck();
     }
 
+    public ngOnMouseOver(header: string): void {
+        this.ngMouseOverHeader = header;
+        this.detectChanges();
+    }
+
+    public ngOnMouseLeave(): void {
+        this.ngMouseOverHeader = '';
+        this.detectChanges();
+    }
```

Comment:
Tracking of mouse over and leave are quite expensive. Please, try to get rid of it. If you need `header` on event `click` you are able to get it with click event and no needs to track it during mouse movements.

Reply:
To show/hide `...` on header you can use `css`, with `::hover` 
```
& .header{
    & mat-icon {
       display: none;
    }
    &:hover {
    & mat-icon {
       display: block;
    }
    }
}
```